### PR TITLE
Add first functional test that uses an account to perform real actions

### PR DESCRIPTION
**Pull request by jerietve**
Update the Azure Pipelines config to download the API token for an account in the integration environment from the key vault. It passes that account to the PowerShell script which invokes Pester. That script logs in the user, and then tests run using that account. A single simple test creates an app, checks it worked and deletes it again.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -51,9 +51,16 @@ jobs:
...
 
+  - task: AzureKeyVault@1
+    inputs:
+      azureSubscription: 'AppCenter - INT'
+      KeyVaultName: 'distributionStoresInt'
+      SecretsFilter: 'ac-cli-account-api-token'
+
   - task: PowerShell@2
     inputs:
       filePath: '$(Build.SourcesDirectory)/test/functional/Main.ps1'
```
Comment (yujshin) [inline azure-pipelines.yml]: Isn't the name "Main.ps1" too generic :-P

File: test/functional/commands/apps/Create.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    # Assert
+    $app.displayName | Should -Be $appDisplayName
+    $app.os | Should -Be $appOs
+    $app.platform | Should -Be $appPlatform
+
+    $newNumberOfApps = (appcenter apps list --output json | ConvertFrom-Json).Count
+    $newNumberOfApps | Should -Be ($previousNumberOfApps + 1)
+
+    # Aftermath
```
Comment (yujshin) [inline test/functional/commands/apps/Create.Tests.ps1]: 'Aftermath' -> 'Cleanup'? Can we do that like AfterEach or something like that?

Comment (jerietve) [reply test/functional/commands/apps/Create.Tests.ps1]: Doesn't start with A! ^^
I'll structure the tests better with before(each) and after(each) code once more tests appear (next week).

File: azure-pipelines.yml [hunk 3]
```diff
@@ -51,9 +51,16 @@ jobs:
       testRunTitle: 'Verify unit test result'
     displayName: 'Verify unit test result'
 
+  - task: AzureKeyVault@1
+    inputs:
+      azureSubscription: 'AppCenter - INT'
+      KeyVaultName: 'distributionStoresInt'
```
Comment (jpeg-ms) [inline azure-pipelines.yml]: Why is this in code?

File: test/functional/Invoke-FunctionalTests.ps1 [hunk 4]
```diff
@@ -1,5 +1,11 @@
 #!/usr/bin/env pwsh
 
+param
+(
+  # The API token used to connect to the App Center integration environment to run the integration tests
```
Comment (jpeg-ms) [inline test/functional/Invoke-FunctionalTests.ps1]: You could use `.SYNOPSIS` and `.PARAMETER` here :)
Also mark as mandatory:
```
[String][Parameter(Mandatory = $true)] $Token
```

**Pull request merged**